### ISSUE TITLE
V2 : Chatbot example custom connection bug

### ISSUE THREAD

**Issue by JosueGauthier** [retejs/rete#633]
Hello, 

I'm using the provided ChatbotV2 example provided by @Ni55aN (https://retejs.org/examples/chatbot/)

I noticed a bug in the management of connections dynamically via the rete-connection-plugin
In the chatbot, a custom connection is declared: Connection. 

When instantiated programmatically, an instance of Connection type is added to the editor: 
```ts
const con6 = new Connection(match, "alternate", send2, "exec");
await editor.addConnection(con6);
```

The problem arises when using the rete-connection-plugin and the following function: 

```ts
connection.addPreset(
    () =>
      new ClassicFlow({
        canMakeConnection(from, to) {
          // this function checks if the old connection should be removed
          const forward = from.side === "output" && to.side === "input";
          const backward = from.side === "input" && to.side === "output";
          const [source, target] = forward
            ? [from, to]
            : backward
            ? [to, from]
            : [];

          if (!source || !target || from === to) return false;

          const con = new Connection(
            editor.getNode(source.nodeId),
            source.key as never,
            editor.getNode(target.nodeId),
            target.key as never
          );

          const sockets = getConnectionSockets(editor, con);
          console.log(sockets);
          console.log(con);

          if (!sockets.source.isCompatibleWith(sockets.target)) {
            log("Sockets are not compatible", "error");
            connection.drop();
            return false;
          }

          return Boolean(source && target);
        }
      })
  );
```

This function, which handles the addition of dynamic connections between nodes, adds not an instance of Connection but an Object that seems to derive from the basic connection class. 

This poses a problem, as it's not possible to work with custom connections. 

An example of the output of the connection list when a connection has been added dynamically: 

```
0: Connection
        sourceOutput: "exec
        targetInput: "exec"
        id: "2490a8e32669d504"
        source: "e8ded87a373fbd15"
        target: "5f1bf2f94bb1f442"
        selected: false
        test: "hhhh"
        <constructor>: "Connec
1: Connection
2: Connection
3: Connection
4: Connection
5: Connection
6: Object 
        id: "8bd4f33f9f367887"
        source: "2d457684c6d2ab3e"
        sourceOutput: "text"
        target: "644018817438b4b5"
        targetInput: "text"

```

Here is a sandbox of the example : [link](https://codesandbox.io/s/rete-js-v2-chatbot-programming-forked-7ltgmt?file=/src/editor.ts)

Any ideas how to solve it ? 

Thanks,

**Comment by Ni55aN** [comment]
thanks for reporting, looks like a bug

**Comment by Ni55aN** [comment]
this is not documented yet. In short, connection plugin uses [`makeConnection`](https://github.com/retejs/connection-plugin/blob/bdce1aec9272137c9cf190b13000a12a577013a0/src/flow/utils.ts#L37) by default. 


You can configure it:

```ts
 new ClassicFlow({
    makeConnection(from, to, context) {
      const [source, target] = getSourceTarget(from, to) || [null, null];
      const { editor } = context;

      if (source && target) {
        editor.addConnection(
          new Connection(
            editor.getNode(source.nodeId),
            source.key as never,
            editor.getNode(target.nodeId),
            target.key as never
          )
        );
        return true;
      }
    }
})
```

updated in ChatbotV2

**Comment by Ni55aN** [comment]
oh, this time it truly proved to be a bug 😅

fixed in rete-vue-render-plugin@2.0.0-beta.25

**Comment by JosueGauthier** [comment]
Thanks for the quick fix ! You are too fast ^^
I close the issue it works.
